cmd/config_test.go

```diff
@@ -83,8 +97,239 @@ func TestConfigCmd(t *testing.T) {
+}
+
+// exp contains the different events or errors we expect our test case to trigger.
+// for space and clarity, we use the fact that by default, all of the struct values are false
+type exp struct {
+	cliError           bool
+	consolidationError bool
+	validationErrors   bool
+	logWarning         bool //TODO: remove in the next version?
+}
+
+// A hell of a complicated test case, that still doesn't test things fully...
+type configConsolidationTestCase struct {
+	options         opts
+	expected        exp
+	customValidator func(t *testing.T, c Config)
+}
+
+var configConsolidationTestCases = []configConsolidationTestCase{
+	// Check that no options will result in 1 VU 1 iter value for execution
+	{opts{}, exp{}, verifyOneIterPerOneVU},
+	// Verify some CLI errors
+	{opts{cli: []string{"--blah", "blah"}}, exp{cliError: true}, nil},
+	{opts{cli: []string{"--duration", "blah"}}, exp{cliError: true}, nil},
+	{opts{cli: []string{"--iterations", "blah"}}, exp{cliError: true}, nil},
+	{opts{cli: []string{"--execution", ""}}, exp{cliError: true}, nil},
+	{opts{cli: []string{"--stage", "10:20s"}}, exp{cliError: true}, nil},
+	// Check if CLI shortcuts generate correct execution values
+	{opts{cli: []string{"--vus", "1", "--iterations", "5"}}, exp{}, verifySharedIters(1, 5)},
+	{opts{cli: []string{"-u", "2", "-i", "6"}}, exp{}, verifySharedIters(2, 6)},
+	{opts{cli: []string{"-u", "3", "-d", "30s"}}, exp{}, verifyConstantLoopingVUs(3, 30*time.Second)},
+	{opts{cli: []string{"-u", "4", "--duration", "60s"}}, exp{}, verifyConstantLoopingVUs(4, 1*time.Minute)},
+	//TODO: verify stages
+	// This should get a validation error since VUs are more than the shared iterations
+	{opts{cli: []string{"--vus", "10", "-i", "6"}}, exp{validationErrors: true}, verifySharedIters(10, 6)},
+	// These should emit a warning
+	{opts{cli: []string{"-u", "1", "-i", "6", "-d", "10s"}}, exp{logWarning: true}, nil},
+	{opts{cli: []string{"-u", "2", "-d", "10s", "-s", "10s:20"}}, exp{logWarning: true}, nil},
+	{opts{cli: []string{"-u", "3", "-i", "5", "-s", "10s:20"}}, exp{logWarning: true}, nil},
+	{opts{cli: []string{"-u", "3", "-d", "0"}}, exp{logWarning: true}, nil},
+	// Test if environment variable shortcuts are working as expected
+	{opts{env: []string{"K6_VUS=5", "K6_ITERATIONS=15"}}, exp{}, verifySharedIters(5, 15)},
+	{opts{env: []string{"K6_VUS=10", "K6_DURATION=20s"}}, exp{}, verifyConstantLoopingVUs(10, 20*time.Second)},
+
+	//TODO: test combinations between options and levels
+	//TODO: test the future full overwriting of the duration/iterations/stages/execution options
+
+	// Just in case, verify that no options will result in the same 1 vu 1 iter config
+	{opts{}, exp{}, verifyOneIterPerOneVU},
+	//TODO: test for differences between flagsets
+	//TODO: more tests in general...
+}
+
+func TestConfigConsolidation(t *testing.T) {
+	logHook := simpleLogrusHook{levels: []log.Level{log.WarnLevel}}
```

Comment:
Please add a comment that this should not be ran in parallel as it touches global state of the logger

Reply:
I have a comment like this below :wink: : But you're right, it should be in the beginning of the test. And to be honest, the logger isn't even the most problematic global state that it touches... :sob: I'd say that the environment variables are a bit more problematic...

Reply:
Done